src/datastore/src/main/java/org/locationtech/geogig/geotools/data/reader/FeatureReaderBuilder.java

```diff
@@ -732,7 +732,15 @@ private SimpleFeatureType resolveMinimalNativeSchema(SimpleFeatureType fullSchem
     private Filter resolveNativeFilter() {
         Filter nativeFilter = this.filter;
         nativeFilter = SimplifyingFilterVisitor.simplify(nativeFilter, nativeSchema);
-        nativeFilter = reprojectFilter(this.filter);
+        nativeFilter = reprojectFilter(nativeFilter);
+
+        String defaultGeometryPName = this.nativeSchema.getGeometryDescriptor().getName().getLocalPart();
```

Comment:
At this point are we shure `this.nativeSchema.getGeometryDescriptor()` is not `null`?